specs/array_intersection_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'minitest/autorun'
+require 'minitest/reporters'
+require_relative '../lib/array_intersection'
+
+describe "array intersection" do
+  describe "basic tests" do
+    describe "test 1" do
+      it "some common elements" do
+        array_1 = [70, 90, 34, 21, 78, 42]
+        array_2 = [21, 67, 54, 29, 42, 30]
+        expected_intersection = [21, 42]
+
+        intersection(array_1, array_2).sort!.must_equal expected_intersection
+      end
+    end
+
+    describe "test 2" do
+      it "some elements common" do
+        array_1 = [92, 2, 5, 4, 1, 89, 90]
+        array_2 = [89, 5, 76, 4, 1, 2]
+        expected_intersection = [1, 2, 4, 5, 89]
+
+        intersection(array_1, array_2).sort!.must_equal expected_intersection
+      end
+    end
+
+    describe "test 3" do
+      it "no elements common" do
+        array_1 = [92, 2, 5, 90]
+        array_2 = [89, 76, 4, 1]
+        expected_intersection = []
+
+        intersection(array_1, array_2).sort!.must_equal expected_intersection
+      end
+    end
+  end
+
+  # if the parameter is an object, check for nil
+  describe "nil object input test" do
+    describe "nil test 1" do
+      it "first parameter nil" do
```

Comment:
I could see this and the other nil test going into the edge cases section.

Reply:
done